src/controllers/productController.js

```diff
@@ -73,7 +73,7 @@ class productController{
     const { name } = req.body;
     try {
 
-      if (req.user.role.roleName !== "vendor") {
+      if (req.user.roleName !== "vendor") {
```

Comment:
Great job with the current code, but I noticed that you're checking the user's role in multiple methods within the `productController` class. To avoid repeating this code and make it more reusable, would you consider creating a middleware function that validates the user's role and returns an error response if the user is not authorized? This way, you can simply call the middleware function in each of the relevant routes and avoid duplicating the same code in multiple places.

If you manage to do that this line will be removed.